### PR TITLE
Building Data Model: first draft for initial discussion of

### PR DESCRIPTION
**Pull request by chicco785**
The implementation is 90% based on GSMA model.
There are few changes introduced by me.

Also, I am not sure it is worth having the BuildingType, my preference
would be to remove it since we have the “category”. An when introducing
json-ld, category labels will be in a reasoner also supporting
hierarchies.

### CODE REVIEW

**Comment by jmcanterafonseca** [comment]
@chicco785 overall comment

In the past we made the mistake of duplicating content already defined by GSMA. I would like to avoid that.

Can we just reference the GSMA document and add only the new attributes that are being proposed, and the rest of artifacts, for instance the JSON Schema, etc. For the GSMA defined content, probably we can just enumerate the fields or, even just show an example in the MD page, but duplicating even descriptions is not so good.

File: Building/Building/doc/spec.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
++ `refBuildingType` : The building's type.
+    + Attribute type: Reference to an entity of type
+      [BuildingType](../../BuildingType/doc/spec.md).
+    + Required
+
++ `category` : One or more categories relevant to the building with choices
+    based on [OpenStreetMap](http://wiki.openstreetmap.org/wiki/Map_Features#Building).
+    Optional but recommended to optimize queries.
+    + Allowed values, one of the following or any other meaningful to the application:
```
[the remark this answers is not in the document]
Comment (chicco785) [reply Building/Building/doc/spec.md]: (y)

**Comment by chicco785** [comment]
hi @jmcanterafonseca 
i think we should keep at least the listing, to avoid people to check different things, in order to be able to use a model. in general, we have then to be careful, because if the model diverge, we will need to document that.

what do you think about the buildingtype? i am for keeping it out.

**Comment by chicco785** [comment]
commit https://github.com/Fiware/dataModels/pull/222/commits/0554be537f08115b55b447b4f6dfc7186d644200 implements @jmcanterafonseca comments

File: Building/Building/doc/spec.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  in the `notes` attribute (remove as well).
+
+* `refSubscriptionService` is not supported, since `SubscriptionService` model
+  is not supported currently.
+
+## Data Model
+
+For a full description of the following attributes refer to GSMA
+[IoT Big Data Harmonised Data Model](https://www.gsma.com/iot/wp-content/uploads/2016/11/CLP.26-v1.0.pdf)
+
```
Comment (jmcanterafonseca) [inline Building/Building/doc/spec.md]: we need to update this reference to point to the latest version of the spec

**Comment by jmcanterafonseca** [comment]
hi @chicco785 

We plan to close v1.0 of Data Models at the end of May, so that the data models can be progressively migrated to NGSI-LD, in parallel with the GSMA IoT Big Data work. 

Do you think you can polish this work by mid May?

Thanks!

File: Building/Building/doc/spec.md (hunk 2 above)
Comment (chicco785) [reply Building/Building/doc/spec.md]: where can i find it?

Comment (chicco785) [reply Building/Building/doc/spec.md]: the link may be wrong actually, this one should the correct one: https://www.gsma.com/iot/wp-content/uploads/2016/06/CLP.26-v4.0.pdf but i couldn't find anything newer

Comment (chicco785) [reply Building/Building/doc/spec.md]: it should be this one, the path ( 2016 ) does not correspond to the pdf data (december 2017). i noticed that some models seems changed (e.g. AirQuality) should be this a concern for us?

**Comment by chicco785** [comment]
@jmcanterafonseca i checked i think there are no pending things (see commits 78e61abbd9a1846d3534c63a175306d33dae94c2 and 506f37cfc2a2cf23a9942bda0ac2bbbed58c7e65)

**Comment by jmcanterafonseca** [comment]
@chicco785 there were some pending comments on my review queue. please check them. once they are fixed we can land it. thanks!

File: Building/Building/doc/spec.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
++ `occupier`
+    + Optional
+
++ `floorsAboveGround`
+    + Optional
+
++ `floorsBelowGround`
+    + Optional
+
++ `refMap`
```
[the remark this answers is not in the document]
Comment (chicco785) [reply Building/Building/doc/spec.md]: is a reference to an uri, that's why i used refMap. not sure though it is the best choice.

File: Building/BuildingOperation/schema.json [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    }
+  ],
+  "required": [
+    "type",
+    "id",
+    "refBuilding",
+    "startDate",
+    "endDate"
+  ]
+}
```
[the remark this answers is not in the document]
Comment (chicco785) [reply Building/BuildingOperation/schema.json]: fix by 1782d74

File: Building/BuildingOperation/example.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  ],
+  "refRelatedBuildingOperation": [
+    "b4fb8bff-1a8f-455f-8cc0-ca43c069f865",
+    "55c24793-3437-4157-9bda-667c9e1531fc"
+  ],
+  "refRelatedDeviceOperation": [
+    "36744245-6716-4a28-84c7-0e3d7520f143",
+    "33b2b713-9223-40a5-87a0-3f80a1264a6c"
+  ]
+}
```
[the remark this answers is not in the document]
Comment (chicco785) [reply Building/BuildingOperation/example.json]: fix by 1782d74

File: Building/Building/example.json [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      "dayOfWeek": "http://schema.org/Monday",
+      "opens": "09:00:00"
+    },
+    {
+      "closes": "17:00:00",
+      "dayOfWeek": "http://schema.org/Wednesday",
+      "opens": "09:00:00"
+    }
+  ]
+}
```
[the remark this answers is not in the document]
Comment (chicco785) [reply Building/Building/example.json]: fix by 1782d74

**Comment by jmcanterafonseca** [comment]
LGTM, thanks

**Pull request merged**